cdap-api/src/main/java/co/cask/cdap/internal/workflow/ProgramWorkflowAction.java

```diff
@@ -40,9 +41,10 @@
 
   private final String name;
   private String programName;
-  private Callable<RuntimeContext> programRunner;
+  private Runnable programRunner;
   private SchedulableProgramType programType;
   private WorkflowContext context;
+  private WorkflowToken token;
```

Comment:
why is this needed here? It also seems that you have a getter for the token, but it never gets set.